### PR TITLE
refactor validation framework to improve user experience and fix several related bugs.

### PR DESCRIPTION
**Pull request by wangmingliang-ms**
[AB#1892770](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1892770): [Test] Admin username keeps validating in simple mode while create SQL
[AB#1893029](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1893029): [Test] OK button disabled after first validation failed and click more settings twice
[AB#1892773](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1892773): [Test] Missing title in Create Resource Group/App Service Plan/Storage Account dialog
[AB#1892692](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1892692): [Test] Pops up NPE error with invalid values for Quota and Retention Period in Create Web App dialog

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AbstractAzureTextInput.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import com.microsoft.azure.toolkit.lib.common.utils.TailingDebouncer;
+
+import javax.annotation.Nullable;
+import javax.swing.*;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+
+public class AbstractAzureTextInput<T> extends ExtendableTextField
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AbstractAzureTextInput.java]: change to `BaseAzureTextInput`? As `AbstractAzureTextInput` may feel like abstract class

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AbstractAzureTextInput.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        AzureValidationInfo.Type.SUCCESS, (i) -> SUCCESS,
+        AzureValidationInfo.Type.ERROR, (i) -> Extension.create(AllIcons.General.BalloonError, i.getMessage(), null),
+        AzureValidationInfo.Type.WARNING, (i) -> Extension.create(AllIcons.General.BalloonWarning, i.getMessage(), null)
+    );
+
+    public AbstractAzureTextInput() {
+        this(null);
+    }
+
+    public AbstractAzureTextInput(@Nullable JTextField comp) {
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AbstractAzureTextInput.java]: is there any cases for null component?

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AzureDialog.java [hunk 3]
```diff
@@ -57,23 +66,13 @@ public void close() {
     @Override
     protected List<ValidationInfo> doValidateAll() {
         final List<AzureValidationInfo> infos = this.getForm().getAllValidationInfos(true);
-        this.setOKActionEnabled(infos.stream().allMatch(AzureValidationInfo::isValid));
+        // this.setOKActionEnabled(infos.stream().allMatch(AzureValidationInfo::isValid));
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AzureDialog.java:69]: When shall we set Ok action enabled?

Comment (wangmingliang-ms) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AzureDialog.java:69]: intellij will handle the enabling/disabling of OK button according to validation results.

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AbstractAzureTextInput.java (hunk 2 above)
Comment (wangmingliang-ms) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AbstractAzureTextInput.java]: the non-arg constructor. null will default to the `getInputComponent()`

**Review by andxu — APPROVED** [review]
LGTM

**Pull request merged**